client/src/components/Sidebar/Sidebar.js

```diff
@@ -29,7 +29,7 @@ const Sidebar = (props) => {
       <CurrentUser />
       <Typography className={classes.title}>Chats</Typography>
       <Search handleChange={handleChange} />
-      {conversations
+      {conversations.slice(0).reverse()
```

Comment:
You shouldn't have to reverse the conversations on the client side; you can retrieve the messages on the server in the correct order once.

Reply:
I'm trying to figure out the correct order for displaying these by changing the order attributes in `Conversation.findAll`, but I'm having no luck. It seems that the outer order `order: [[Message, "createdAt", "DESC"]],` must be kept as `DESC` to be able to change the order of the conversations at all. However, doing so means I cannot change the order of the inner `{ model: Message, order: [Message, "createdAt", "DESC"] }` as it does not seem to have any effect whatever I change this to. The result of this is the correct conversation order but the messages are upside down for each chat (old messages at the bottom, new at the top). 

I'm stuck on this because if I change the outer order to `order: [[Message, "createdAt", "ASC"]],`, the conversation order is not preserved.

```js 
...
order: [[Message, "createdAt", "DESC"]],
      include: [
        { model: Message, order: ["createdAt", "DESC"] }, //this order has no effect
        ...
```

Reply:
I've decided to go with this solution for now, please let me know if there is a way to do it directly in the findAll.

```js
for (let i = 0; i < conversations.length; i++) {
      const convo = conversations[i];
      const convoJSON = convo.toJSON();
      convoJSON.messages = convoJSON.messages.reverse();
      ...
```

Reply:
For efficiency, it'd be better to get the messages in the correct order and then sort the conversations by the last message's `createdAt` timestamp.